demo/src/main/java/io/weet/demo/services/UserService.java

```diff
@@ -49,4 +49,30 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
         return new User(user.getEmail(), user.getPassword(), authorities);
     }
 
+    //add allergen and updates it to the user object in the database
+
+    public UserModel addAllergen(String email, String allergenName){
+
+        UserModel user = userRepository.findByEmail(email); //gets the user in question
+
+        user.addAllergy(allergenName); //updates user's list of allergens by adding it
+        userRepository.save(user);
+
+        return user;
+
+    }
+
+    public UserModel deleteAllergen(String email, String allergenName){
+        
+        UserModel user = userRepository.findByEmail(email);
+
+        user.deleteAllergy(allergenName);
+        userRepository.save(user);
+        
+        return user;
+
+    }
```

Comment:
This isn't needed. The user object should load the allergen data from DB so that it isn't manually added.